cmake/tribits/core/package_arch/TribitsDefineStandardCompileVars.cmake

```diff
@@ -56,15 +56,15 @@ MACRO(TRIBITS_DEFINE_STANDARD_COMPILE_FLAGS_VARS  ENABLE_SHADOWING_WARNINGS)
   # Setup and general flags
   #
 
-  SET(GENERAL_BUILD_FLAGS) # Applies to all builds, period
+  SET(GENERAL_BUILD_FLAGS "-pipe") # Applies to all builds, period
```

Comment:
So I'm concerned about changing this file.  I believe this is actually part of TriBITS itself.  I'm not sure what downstream effects this could potentially have... @bscollin @stimpsonsg care to comment?  You guys know a bit more about tribits than I do

Reply:
Is it possible to set these flags (or modify them) via commandline instead?

Reply:
> To questions to clarify:
> 
> 1. Did you succeed in getting this flag to filter down into Trilinos as well?  I assume using the "general flags" instead of the "cxx flags" might accomplish that, but want to double check.
> 2. If the machine does not have enough memory, does the behavior automatically just revert to as if the `-pipe` flag were not used?

1) Yes it does filter down to trilinos doing it this way. Adding it to the general flags obviates the need to change more files to add the -pipe flag such as Futility_Configurations

2) That is a serious question. Unfortunately the gcc manual does not specify. It either page files or reverts to intermediates, I don't know which actually happens.

Reply:
We've had several offline conversations about this, but just for the sake of thoroughness, here's where we've arrived:

1. The flag can be set via commandline.  We're going to just add it to some of the CI build scripts for now rather than modifying any tribits/cmake code.
2. If the machine runs out of memory with this option, it will simply start using swap space (tested locally on my laptop).  However, the amount of memory use for a build isn't that crazy.  Something else had to be pushing it close to the edge for a single build to run out of memory (I saw < 8 GB as the highwater mark for my build).